src/injected/all-frame-runner.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HTMLElementUtils } from 'common/html-element-utils';
+import { WindowUtils } from 'common/window-utils';
+import { FrameMessenger } from 'injected/frameCommunicators/frame-messenger';
+import {
+    CommandMessage,
+    CommandMessageResponse,
+} from 'injected/frameCommunicators/respondable-command-message-communicator';
+
+export interface AllFrameRunnerTarget<T> {
+    start: () => void;
+    teardown: () => void;
+    transformChildResultForParent: (fromChild: T, messageSourceFrame: HTMLIFrameElement) => T;
+    setResultCallback: (reportResults: (payload: T) => Promise<null>) => void;
+}
+
+/*
```

Comment:
note for reviewer: most of the iframe code is moved over from tab-stops-listener